apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -301,16 +287,15 @@ export class SendMessage {
   }
 
   @Instrument()
-  private async filterPreferredChannels(job: JobEntity): Promise<boolean> {
+  private async filterPreferredChannels(command: SendMessageCommand): Promise<boolean> {
+    const { job } = command;
+
     const template = await this.getNotificationTemplate({
       _id: job._templateId,
       environmentId: job._environmentId,
     });
-    if (!template) {
-      throw new PlatformException(`Notification template ${job._templateId} is not found`);
-    }
 
-    if (template.critical || this.isActionStep(job)) {
+    if (template?.critical || this.isActionStep(job)) {
```

Comment:
Template can be undefined at this point of the flow now.

Reply:
That's correct for V2 Preferences, however we still need this check in place to cater to V1 Workflows. I will add a code comment to make that clear.